### ISSUE TITLE
Make `Semigroup` instance more composable

### ISSUE THREAD

**Issue by unclechu** [massysett/rainbow#11]
When I write `putChunkLn (fore red "foo" <> fore blue "bar")` I'd expect `foo` being red and `bar` being blue while in real life I get the whole string being blue.

![Screenshot from 2019-12-11 04-13-42](https://user-images.githubusercontent.com/799353/70713607-1008e000-1cef-11ea-9100-5ed5d5396f59.png)

It doesn't violates the `Semigroup` laws but it's not intuitive behaviour by my opinion and leads to a confusion. I'd like `Semigroup Chunk` instance being fixed or to have a wrapper provided by the rainbow library like this one I wrote for myself:

```haskell
newtype Chunks = Chunks { fromChunks :: [Chunk] } deriving (Eq, Ord, Show)

instance Semigroup Chunks where
  Chunks xs <> Chunks ys = Chunks (xs <> ys)

instance Monoid Chunks where
  mempty = Chunks mempty

instance IsString Chunks where
  fromString x = Chunks [fromString x]
```

And I have some helpers like this:
```haskell
colorize :: (Chunk -> Chunk) -> Chunks -> Chunks
```
And a usage example would be:
```haskell
poutChunksLn $ fromChunks $
  colorize (fore red) "foo" <> colorize (fore blue) "bar"
```

**Comment by massysett** [comment]
By design each Chunk has a single set of colors and effects.  You're combining two Chunks into one, and the documented result of this is that it's going to take on the rightmost color.  To do what you want, try `putChunksLn [fore red "foo", fore blue "bar"]`.  This is documented in the "Rainbow" module.

I'm not sure the `Semigroup` or `Monoid` instances on `Chunk` are useful though, so I might remove them as this would eliminate this sort of confusion.

**Comment by unclechu** [comment]
@massysett Yes, I'd vote for removing those instances in this case since they are rather confusing than helpful.
